### ISSUE TITLE
Type Acquisition does not work on packages that include their own typings

### ISSUE THREAD

**Issue by matthew-e-brown** [microsoft/TypeScript#50276]
When using JavaScript files in a more "traditional" environment, where they are included directly in the browser with `<script src="..."></script>`, it's nice to still get type checking. `checkJs` and `jsconfig.json` are very nice for this. Sadly, things get a little hairy when you want to start working with libraries that are published on a CDN.

The normal flow for this is to simply add an extra `<script>` or `<link>` tags to the head of your document, and then refer to a global function or namespace (the most obvious example being JQuery's `$`). This works great with the `typeAcquisition` feature of `jsconfig.json`. However, that only seems to be the case for libraries that do no bundle their own type definitions: when VS Code attempts to locate types, it seems to only want to pull them from `@types/*`.

I have scoured the ["using JavaScript" docs](https://code.visualstudio.com/docs/nodejs/working-with-javascript) as carefully as I can manage, but shy of making my own `index.d.ts` and manually declaring things I want to use, I cannot figure out how to get VS Code to give me these types.

From all the searching I did, I found that this problem was also reported in microsoft/vscode#58275, which went stale.

----

Type: <b>Bug</b>

1.  Create a new project, and create a `jsconfig.json` and `.js` file.
2.  Enable type acquisition, and include a library found on `@types/*`; for example, [Papaparse](https://www.npmjs.com/package/papaparse):
    ```jsonc
    // jsconfig.json
    {
      "typeAcquisition": {
        "enable": true,
        "include": [
          "papaparse"
        ]
      }
    }
    ```
3. See that types are correctly inferred globally, like when Papaparse's `<script>` tag would be included in the `<head>` of the document.  \
   ![](https://user-images.githubusercontent.com/43387454/184153747-9e5ca2a0-da17-42a0-a2e5-8bbba4cd176f.png)
4.  Add a library that also declares global variables for use with an external script, but includes its own type declarations; for example, [tippy.js](https://www.npmjs.com/package/tippy.js):
    ```jsonc
    // jsconfig.json
    {
      "typeAcquisition": {
        "enable": true,
        "include": [
          "papaparse",
          "tippy.js"
        ]
      }
    }
    ```
5.  See that types are not inferred globally, despite tippy.js's `index.d.ts` including global declarations:  \
   ![](https://user-images.githubusercontent.com/43387454/184153835-b248bc11-4dc7-4a53-9698-ff8fceb04502.png)
    ```ts
    // curl -o- $(npm v tippy.js dist.tarball) > tippy.js.tar.gz 2> /dev/null && tar -zxf tippy.js.tar.gz && sed -n 158,160p package/index.d.ts

    declare const tippy: Tippy;

    ```

----

- Does this issue occur when all extensions are disabled: Yes
- VS Code version: Code 1.70.1 (6d9b74a70ca9c7733b29f0456fd8195364076dda, 2022-08-10T06:08:33.642Z)
- OS version: Windows_NT x64 10.0.19044
- Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2808)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.86GB (5.62GB free)|
|Process Argv|. --crash-reporter-id a60d9f3f-3deb-45a9-bd6f-ee70f55318fb|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vstes627:30244334
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30542924
vsaa593:30376534
pythonvs932:30410667
wslgetstarted:30449410
cppdebug:30492333
pylanb8912:30529769
vsclangdf:30486550
c4g48928:30535728
d7fjc500:30543455
dsvsc012:30540252

```

</details>

**Comment by mjbvz** [comment]
`typeAcquisition` doesn't sound like the correct setting. Have you tried installing the package into your project: `npm install tippy.js`?

**Comment by RyanCavanaugh** [comment]
Enabling `typeAcquisition` won't do anything for this scenario. TS/JS/VS Code will never automatically install a package itself -- doing so would be potentially very impactful (e.g. some npm packages will load several hundred megabytes of data).

If the package is installed into the project's local node_modules, it'll get picked up automatically.

**Comment by matthew-e-brown** [comment]
> @mjbvz  
> Have you tried installing the package into your project: `npm install tippy.js?`

I was hoping I didn't have to do that: if I can use one package's types without having to introduce a `package.json` to this project, I hoped I could use any package's types without having to do so.

Moving to a bundler/build step isn't feasible/desired for this project, so we don't have a `package.json` at all at the moment. That makes this feature feel incomplete to me: if I have to install things as `devDependencies` for my `ts-check` to work at all, it's odd to me that that requirement is only for some libraries&mdash;at that point I may as well just install **all** the `@types/*` packages manually and never use `typeAcquisition`.

I guess it's simply not intended for this use-case.

----

> @RyanCavanaugh  
> TS/JS/VS Code will never automatically install a package itself -- doing so would be potentially very impactful (e.g. some npm packages will load several hundred megabytes of data).

That makes sense. I figured that me *manually* specifying packages in `include` would allow me to tell it to basically do what an `npm install` would do.

...I guess at that point I basically have a mini-`package.json` and may as well just one a real one 😅

----

Oh, well. Thank you, to you both.

**Comment by thomthom** [comment]
> Enabling `typeAcquisition` won't do anything for this scenario. TS/JS/VS Code will never automatically install a package itself -- doing so would be potentially very impactful (e.g. some npm packages will load several hundred megabytes of data).
> 
> If the package is installed into the project's local node_modules, it'll get picked up automatically.

The "Working With JavaScript" docs says (https://code.visualstudio.com/docs/nodejs/working-with-javascript):

> Type declaration files are automatically downloaded and managed by Visual Studio Code for packages listed in your project's package.json *or that you import into a JavaScript file.*

(Emphasis mine). That bit had me thinking that listing the libs I wanted type hints for in `typeAcquisition.include` was enough. I realize this thread is 2.5 years old, so I'm wondering is this still the case?
